### PR TITLE
[ch64] As a user, I can see About Us screen

### PR DESCRIPTION
**Pull request by Raksani**
[https://app.clubhouse.io/cobda/story/64/as-a-user-i-can-see-about-us-screen](url)

## Why?

This **PR** is the entry point for the `About Us` screen. And accessible through the `Footer` component. 

## Changes

- Create `About Us` screen.
- Add `About us` link on the `Footer`.

## Screenshot(s)

**Before**
105196496-478ec080-5b6e-11eb-8bb6-57f55272deca

**After**
About us link appears on Footer. And About Us screens was created.
Screenshot 2564-01-22 at 23 11 19

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "node-sass": "4.14.1",
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
+    "react-router-dom": "^5.2.0",
```
Comment (LilBank) [inline package.json:15]: Quick question, I thought we already have this dependency. Or at first it wasn't listed on your **PR**? 🤔 @kimvcp

File: src/screens/AboutUs/styles.scss [hunk 2]
```diff
@@ -0,0 +1 @@
+@import '../../styles/main.scss';
```
Comment (LilBank) [inline src/screens/AboutUs/styles.scss]: Please add newline.
```suggestion
@import '../../styles/main.scss';

```

File: src/components/Footer/index.tsx [hunk 3]
```diff
@@ -1,6 +1,13 @@
...
 
 export default function Footer() {
-  return <div className="footer-container">Footer</div>;
+  return (
+    <div className="footer-container">Footer
+        <div className="footer-item">
+        <Link to="/about-us">About Us</Link>
+      </div>
+    </div>
+  );
```
Comment (LilBank) [inline src/components/Footer/index.tsx]: Indention.
```suggestion
  return (
    <div className="footer-container">
      Footer
      <div className="footer-item">
        <Link to="/about-us">About Us</Link>
      </div>
    </div>
  );
```

File: src/components/Footer/styles.scss [hunk 4]
```diff
@@ -9,3 +9,9 @@
     height: rem(72px);
   }
 }
+
+.footer-item {
+  font-size: 16px;
+  font-weight: 500;
+  line-height: normal;
+}
```
Comment (LilBank) [inline src/components/Footer/styles.scss]: How about approach it sass style?

```suggestion
@import '../../styles/main.scss';

.footer {
  height: rem(32px);
  background-color: $secondary;
  
  @include flex-center();
  @include respond-above(md) {
     height: rem(72px);
   }
   
  &__item {
    font-size: 16px;
    font-weight: 500;
    line-height: normal;
  }
}
```

Comment (LilBank) [reply src/components/Footer/styles.scss]: Source: https://css-tricks.com/sass-style-guide/

**Review by LilBank — CHANGES_REQUESTED** [review]

**Comment by LilBank** [comment]
Your previous commits doesn't contain `[ch]` syntax. Please input this so the PR integrates with clubhouse.

**Comment by Raksani** [comment]
> Your previous commits doesn't contain `[ch]` syntax. Please input this so the PR integrates with clubhouse.
Apologize for my mistakes. I will fix them ASAP.

**Comment by Raksani** [comment]
> Your previous commits doesn't contain `[ch]` syntax. Please input this so the PR integrates with clubhouse.

Apologize for my mistakes. I will fix them ASAP.

File: package.json (hunk 1 above)
Comment (kimvcp) [reply package.json:15]: In my PR, this `react-router-dom` was installed only the `@types` scope for Typescript. However, we could have them exisiting together.

Comment (kimvcp) [reply package.json:15]: I just found that it is the best practice to have both `react-router-dom` and `@types/react-router-dom` installed.
Ref: https://www.sitepoint.com/react-with-typescript-best-practices/

File: src/router/index.tsx [hunk 5]
```diff
@@ -15,6 +16,9 @@ export default function Routes() {
         <Route exact path="/">
           <Home />
         </Route>
+        <Route exact path="/about-us">
```
Comment (kimvcp) [inline src/router/index.tsx:19]:
```suggestion
        <Route path="/about-us">
```

Quick suggestion, this `<Route>` does not need to have `exact` property.  
Ref: https://reactrouter.com/web/api/Route

**Comment by LilBank** [comment]
> Apologize for my mistakes. I will fix them ASAP.

In future, no need to apologize. Human mistake is inevitable. Just notify my comment with an emoji and fix it 👍

**Comment by LilBank** [comment]
In changes, no need to be say **Create SCSS files for About Us screen** since these files will be seen in changes. In changes you should tell the main changes which it should only be **Create About us screen**/**Add link About us screen in footer**.

File: src/router/index.tsx (hunk 5 above)
Comment (Raksani) [reply src/router/index.tsx:19]: To be clear, why we only put `exact` only for path `/`.

File: src/components/Footer/index.tsx [hunk 6]
```diff
@@ -1,6 +1,14 @@
 import React from 'react';
+import { Link } from 'react-router-dom'
 import './styles.scss';
 
 export default function Footer() {
-  return <div className="footer-container">Footer</div>;
+  return (
+    <div className="footer-container">
+      Footer
+      <div className="footer-item">
```
Comment (LilBank) [inline src/components/Footer/index.tsx:9]: There are no such `footer-container` and `footer-item`. If you see `Footer/styles.scss`, there's only `.footer` and `footer__item`. 🧐

File: src/components/Footer/index.tsx [hunk 7]
```diff
@@ -1,6 +1,14 @@
 import React from 'react';
+import { Link } from 'react-router-dom'
```
Comment (LilBank) [inline src/components/Footer/index.tsx:2]: Typoo! 🤓
```suggestion
import { Link } from 'react-router-dom';
```

File: src/components/Footer/styles.scss [hunk 8]
```diff
@@ -9,3 +9,20 @@
...
+  @include respond-above(md) {
+     height: rem(72px);
+   }
+   
+  &__item {
+    font-size: 16px;
+    font-weight: 500;
+    line-height: normal;
+  }
+}
```
Comment (LilBank) [inline src/components/Footer/styles.scss:28]: According to the solution I've guided, there's no need for `.footer-container`. 🙅🏻‍♂️

Comment (LilBank) [reply src/components/Footer/styles.scss:28]: Seems like you've copy/paste the solution without any understandings. Please make sure you do understand what the code does. If you don't understand then raise a your hand without any hesitation and we'll help you. 🙇🏻‍♂️

File: src/screens/AboutUs/index.tsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+import './styles.scss';
+
+export default function AboutUs() {
+  return (
+    <>
+      <header>
+        <Navbar />
+      </header>
+      <main>
+        <Sidebar/>
```
Comment (LilBank) [inline src/screens/AboutUs/index.tsx:14]: Typo 🥸
```suggestion
        <Sidebar />
```

**Review by LilBank — CHANGES_REQUESTED** [review]

**Comment by LilBank** [comment]
Screen Shot 2564-01-23 at 20 49 22
FYI, you don't have to specify which file you've edited since the commit message will be attached to the file that changes within that commit. Also commit message doesn't have a period(.), please remove them with `git rebase -i`.
Source: https://www.youtube.com/watch?v=ElRzTuYln0M
Please kindly watch this since this is hard 🥰

**Comment by Raksani** [comment]
Due to many conflicts, I have to close this pull request and re-build this branch.

**Pull request closed without merging**